cmd/nic-configuration-daemon/main.go

```diff
@@ -59,14 +67,29 @@ func main() {
 	}
 
 	hostUtils := host.NewHostUtils()
-	hostManager := host.NewHostManager(hostUtils)
+	hostManager := host.NewHostManager(nodeName, hostUtils)
+	maintenanceManager := maintenance.New(mgr.GetClient(), hostUtils, nodeName, namespace)
```

Comment:
We should follow the same pattern as in L79

Reply:
I would rather not because that way we hide the implementation inside the package